packages/widget-v2/src/pages/SwapExecutionPage/SwapExecutionPage.tsx

```diff
@@ -51,8 +53,43 @@ export const SwapExecutionPage = () => {
     2: "pending",
   });
 
+  const { mutate, isPending, isSuccess } = useAtomValue(skipSubmitSwapExecutionAtom);
+
+  const swapExecutionState = useMemo(() => {
+    if (!chainAddresses) return;
+    const requiredChainAddresses = route?.requiredChainAddresses;
+    if (!requiredChainAddresses) return;
+    const allAddressesSet = requiredChainAddresses.every(
+      (_chainId, index) => chainAddresses[index]?.address
+    );
+
+    const lastChainAddress = chainAddresses[requiredChainAddresses.length - 1]?.address;
+
+    if (!isPending) {
+      if (allAddressesSet) {
+        return SwapExecutionState.ready;
+      }
+      if (!lastChainAddress) {
+        return SwapExecutionState.destinationAddressUnset;
+      }
+      return SwapExecutionState.recoveryAddressUnset;
+    }
```

Comment:
do we need to wrap these in !isPending? can we do something like this?

if (!isPending) return '';
if (allAddressesSet) {
  return SwapExecutionState.ready;
}
if (!lastChainAddress) {
  return SwapExecutionState.destinationAddressUnset;
}
return SwapExecutionState.recoveryAddressUnset;

Reply:
it's easier to read if we wrap it in !isPending, because those state only happens before submit